sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/CosmosThrottled.java

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables;
+
+import com.azure.core.exception.AzureException;
+import com.azure.data.tables.implementation.AzureTableImpl;
+import com.azure.data.tables.implementation.models.TableServiceErrorException;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public abstract class CosmosThrottled<T> {
+    protected final T client;
+    protected final boolean isPlaybackMode;
+
+    protected CosmosThrottled(T client, boolean isPlaybackMode) {
+        this.client = client;
+        this.isPlaybackMode = isPlaybackMode;
+    }
+
+    public abstract boolean isCosmos();
+
+    public void runVoid(Consumer<T> action) {
+        run(c -> {
+            action.accept(c);
+            return null;
+        });
+    }
+
+    public T getClient() {
+        return client;
+    }
+
+    public <TResult> TResult run(Function<T, TResult> action) {
+        if (!isCosmos()) {
+            return action.apply(client);
+        }
+
+        int retryCount = 0;
+        int delay = 1500;
+        while (true) {
+            try {
+                return action.apply(client);
```

Comment:
Does this work for async? The apply method would return a Mono<Void> but it would not actually be invoked until someone subscribed to it.

I think a more reactive approach is required to keep retrying depending on the error. there is a retryWhen operator.

Reply:
Turns out that this was actually unneeded, I just needed to configure the retrypolicy to allow more retries & backoff time.